modules/storages/app/common/storages/peripherals/oauth_configurations/one_drive_configuration.rb

```diff
@@ -63,8 +64,8 @@ def extract_origin_user_id(rack_access_token)
           ).raise_for_status.json['id']
         end
 
-        def scope
-          DEFAULT_SCOPES
+        def scope(user:)
+          user.nil? ? %w[https://graph.microsoft.com/.default] : DEFAULT_SCOPES
```

Comment:
I think we can always use the `.default` one, as it will grant the permissions allowed by the Azure Application.

It might be worth a test.